src/client/credentials/authorization.js

```diff
@@ -149,7 +149,7 @@ var Authorization = SparkBase.extend({
         }.bind(this))
     ])
       .then(function revokeSupertoken() {
-        return supertoken.revoke()
+        return supertoken && supertoken.revoke()
```

Comment:
i think sometimes we call logout() when the user is already logged out. this happens sometimes when we get a sideboarded link and we need to log out the previous user out, but the device and supertoken is already unregistered/revoked